### PR TITLE
(GH-2633) Add Bolt Task directory to PSModulePath

### PR DESCRIPTION
**Pull request by jpogran**
This commit adds the bolt task target directory to the PSModulePath environment variable for the currently executing session. This will allow Bolt tasks to ship powershell modules that can be automatically imported during runtime instead of having to calculate relative paths.

To use this feature add any powershell module to the `files` metadata in the task.json file:

```
"files": [
  "<BOLT MODULE NAME>/files/<PWSH MODULE NAME>/"
],
```

Then inside your Bolt PowerShell task start using the content from that PowerShell module without having to write `Import-Module` statements with fully qualified paths to the temp directory Bolt operates in.

### CODE REVIEW

File: documentation/writing_tasks.md
Comment (hestonhoffman) [inline documentation/writing_tasks.md:984]:
```suggestion
If using PowerShell 3.0, add an import statement here:
```

File: documentation/writing_tasks.md
Comment (hestonhoffman) [inline documentation/writing_tasks.md:991]:
```suggestion
> **Note**: You don't have to write an import statement to load your PowerShell module. 
  Bolt automatically loads the module by adding your task directory to 
  `$env:PSModulePath`.
```

File: lib/bolt/shell/powershell.rb
Comment (beechtom) [inline lib/bolt/shell/powershell.rb:279]: See comment for `#append_ps_module_path`. If making that change, this would be updated to take `dir` and `extra_files` as parameters.

File: documentation/writing_tasks.md
Comment (beechtom) [inline documentation/writing_tasks.md:969]:
```suggestion
  "parameters": {
    "name": {
      "description": "Name or list of SCCM Trigger to run",
      "type": "String"
    }
  }
}
```

Metadata needs to list the parameters the task accepts.

File: documentation/writing_tasks.md
Comment (hestonhoffman) [inline documentation/writing_tasks.md:995]:
```suggestion
> **Note**: If you're using a version of PowerShell more recent than 3.0, 
  you don't have to write an import statement to load your PowerShell module. 
  Bolt automatically loads the module by adding your task directory to 
  `$env:PSModulePath`.
```

File: documentation/writing_tasks.md
Comment (beechtom) [inline documentation/writing_tasks.md:967]:
```suggestion
  "files": ["powershell_task_helper/files/BoltPwshHelper/"],
```

**Review by beechtom — COMMENTED** [review]
It looks like the example task is incomplete.

File: documentation/writing_tasks.md
Comment (jpogran) [reply documentation/writing_tasks.md:961]: This was a copy pasta from the ruby/python example so we have more to change

File: lib/bolt/shell/powershell/snippets.rb
Comment (jpogran) [reply lib/bolt/shell/powershell/snippets.rb:75]: If we add the task directory itself, PowerShell traverses the tree to find any module there. So any powershell module included in the puppet module that the task is in is available to autoload. We don't have to add directories specifically. This works like Puppet or Bolt in that respect.

We could be more restrictive if we only add the `files` directory, then only content in that folder is available for autoload.

Comment (beechtom) [reply lib/bolt/shell/powershell/snippets.rb:75]: Makes sense. Should the above be `Split-Path "#{directory}" -Parent` in that case? Currently it's just adding the `tasks/` directory to the path, which won't include the extra files listed in the metadata.

Also, do we eventually want to support this behavior for files from other modules (like the powershell task helper)?

File: documentation/writing_tasks.md
Comment (lucywyman) [inline documentation/writing_tasks.md:973]:
```suggestion
`<PROJECT DIRECTORY>/tasks/mytask.{json,ps1}`.
```
Asked about this on one of Tom's PRs too - we standardized on this a few months ago, but I'm not sure if `PROJECT` is better....

File: documentation/writing_tasks.md
Comment (beechtom) [inline documentation/writing_tasks.md:1010]: We should use the complete task and metadata example from https://github.com/jpogran/puppetlabs-powershell_task_helper/tree/pwsh-content/examples , since this example doesn't produce the output shown below.

**Review by beechtom — APPROVED** [review]
Just needs an updated example, but otherwise lgtm 👍

File: documentation/writing_tasks.md
Comment (jpogran) [reply documentation/writing_tasks.md]: Updated

**Review by beechtom — APPROVED** [review]
Do we want to add the powershell_task_helper module to the Puppetfile in this PR or a separate one? Either works.

**Comment by jpogran** [comment]
@beechtom I can add it, but the powershell_task_helper hasn't been pushed to the forge yet so that needs to be done before we could merge this PR

**Comment by beechtom** [comment]
Let's go ahead and merge this and then add the module separately, since ci-job-configs also needs to be updated.

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**
